internal/pkg/pluginengine/change.go

```diff
@@ -40,27 +40,31 @@ const (
 // GetChangesForApply takes "State Manager" & "Config" then do some calculate and return a Plan.
 // All actions should be execute is included in this Plan.changes.
 func GetChangesForApply(smgr statemanager.Manager, cfg *configloader.Config) ([]*Change, error) {
-	return getChanges(smgr, cfg, CommandApply)
+	return getChanges(smgr, cfg, CommandApply, false)
 }
 
 // GetChangesForDelete takes "State Manager" & "Config" then do some calculation and return a Plan to delete all plugins in the Config.
 // All actions should be execute is included in this Plan.changes.
-func GetChangesForDelete(smgr statemanager.Manager, cfg *configloader.Config) ([]*Change, error) {
-	return getChanges(smgr, cfg, CommandDelete)
+func GetChangesForDelete(smgr statemanager.Manager, cfg *configloader.Config, isForce bool) ([]*Change, error) {
+	return getChanges(smgr, cfg, CommandDelete, isForce)
 }
 
-func getChanges(smgr statemanager.Manager, cfg *configloader.Config, commandType CommandType) ([]*Change, error) {
+func getChanges(smgr statemanager.Manager, cfg *configloader.Config, commandType CommandType, isForce bool) ([]*Change, error) {
 	if cfg == nil {
 		return make([]*Change, 0), nil
 	}
-
+	log.Debug("isForce:", isForce)
 	// calculate changes from config and state
 	var changes []*Change
 	var err error
 	if commandType == CommandApply {
 		changes, err = changesForApply(smgr, cfg)
 	} else if commandType == CommandDelete {
-		changes = changesForDelete(smgr, cfg)
+		if isForce {
```

Comment:
`changes = changesForDelete(smgr, cfg, isForce)`

I think it's better to write it like the above; of course this is just my personal opinion and it doesn't have to be write this way. you can decide for yourself.

Reply:
I think in this format,  if isForce=true, 'change'  will be generated twice